mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -160,7 +161,43 @@ public boolean hasChildren(Mailbox mailbox, char delimiter) throws MailboxExcept
     }
 
     @Override
-    public MailboxId save(Mailbox mailbox) throws MailboxException {
+    public MailboxId create(Mailbox mailbox) throws MailboxException {
+        Preconditions.checkArgument(mailbox.getMailboxId() == null, "A mailbox we want to create should not have a mailboxId set already");
+
+        MaildirId maildirId = MaildirId.random();
+        MaildirFolder folder = maildirStore.createMaildirFolder(mailbox);
+
+        if (!folder.exists()) {
+            boolean success = folder.getRootFile().exists();
+            if (!success) {
+                success = folder.getRootFile().mkdirs();
```

Comment:
please don't reuse success variable, it makes the code too complicated

Reply:
Moved to an other PR -> #3108, code of create for other implementation is not necessary for just create adoption on Cassandra (which is the primary goal of the ticket)